services/src/main/java/org/fao/geonet/api/records/MetadataWorkflowApi.java

```diff
@@ -692,6 +697,7 @@ public List<MetadataStatusResponse> getWorkflowStatusByType(
         Integer size,
         HttpServletRequest request) throws Exception {
         ServiceContext context = ApiUtils.createServiceContext(request);
+        checkUserProfileToViewMetadataHistory(context.getUserSession());
```

Comment:
Doesn't this code replace the code below that starts with 

`if (profile != Profile.Administrator) {`

If so then I would expect that if statement to be removed.

Reply:
Sorry - I guess it is not a full replacement since there is other logic in there.

I wonder if we would need to add more options

Registered User, **All Editor**, Record Editors, Reviewer or Administrator

In this case "Record Editors, Reviewer or Administrator" would be using the old existing logic

All Editor - would be anyone who is an editor and can view the record
Registered User - would be anyone who is registered and can view the record
So for the last 2 instead of using ` ApiUtils.canEditRecord`, you could use  `ApiUtils.canViewRecord`

Reply:
yeah, i think in this case, the other community members has less impact. Their implementation can be as is. So we introduce the logic on our end. So all we need to do is entering the original logic by 
if user.profile!=RegisterUser or Admin, then do blah blah

Reply:
I added some logic for accessLevel=RegisterdUser, if this user can see the metadata record, then this history API should allow them to see the result. Otherwise, if they are set to Editor level, it will use the old logic to check if this person can edit this record or not. In this case, it's backward compactable.